lms/static/js/commerce/track_ecommerce_events.js

```diff
@@ -0,0 +1,34 @@
+/**
+ *
+ * A library of helper functions to track ecommerce related events.
+ *
+ */
+(function(define) {
+    'use strict';
+    define([], function() {
+        var trackUpsellClick = function(linkName, optionalAttrs) {
+            var eventAttrs = {linkName: linkName};
+            var allowedAttrs = ['linkType', 'pageName', 'linkCategory'];
```

Comment:
We'll also need to send the userId and courseId- does this come through into your event context automatically? Or does it need to be added in?

Reply:
The userID should be included by default; I'm not sure if the courseID is included by default but I think the referrer is, which should have the name/id of the course in the URL, so hopefully that will be good enough for the data analysis?